tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/views/stack_trace/stack_trace_container.ts

```diff
@@ -30,16 +33,53 @@ import {StackFrameForDisplay} from './stack_trace_component';
   selector: 'tf-debugger-v2-stack-trace',
   template: `
     <stack-trace-component
+      [stackTraceType]="stackTraceType$ | async"
+      [originOpInfo]="originOpInfo$ | async"
       [stackFramesForDisplay]="stackFramesForDisplay$ | async"
       (onSourceLineClicked)="onSourceLineClicked($event)"
     ></stack-trace-component>
   `,
 })
 export class StackTraceContainer {
+  readonly stackTraceType$ = this.store.pipe(select(getCodeLocationFocusType));
+
+  readonly originOpInfo$ = this.store.pipe(
+    select(
+      createSelector(
+        getCodeLocationFocusType,
+        getFocusedExecutionData,
+        getFocusedGraphOpInfo,
+        (codeLocationFocusType, executionData, graphOpInfo) => {
+          if (codeLocationFocusType === null) {
+            return null;
+          }
+          if (codeLocationFocusType === CodeLocationType.EXECUTION) {
+            if (executionData === null) {
+              return null;
+            }
+            return {
+              opType: executionData.op_type,
+              opName: null,
+            };
+          } else {
+            // This is CodeLocationType.GRAPH_OP_CREATION.
+            if (graphOpInfo === null) {
+              return null;
+            }
+            return {
+              opType: graphOpInfo.op_type,
+              opName: graphOpInfo.op_name,
+            };
+          }
+        }
```

Comment:
Is there a way to abstract out these dirty details of `focusType` here and in `getFocusedStackFrames`? :\ I really do not wish to make view aware of this detail if possible.

Reply:
Follow-up: the new selector is now called `getCodeLocationOrigin()`, as mentioned in the other reply comment.